akka-cluster/src/multi-jvm/scala/akka/cluster/RestartNode2Spec.scala

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.cluster
+
+import scala.collection.immutable
+import scala.concurrent.duration._
+
+import akka.actor.Actor
+import akka.actor.ActorSystem
+import akka.actor.Address
+import akka.actor.Deploy
+import akka.actor.Props
+import akka.actor.RootActorPath
+import akka.cluster.ClusterEvent._
+import akka.cluster.MemberStatus._
+import akka.remote.AddressUidExtension
+import akka.remote.testkit.MultiNodeConfig
+import akka.remote.testkit.MultiNodeSpec
+import akka.testkit._
+import com.typesafe.config.ConfigFactory
+
+object RestartNode2SpecMultiJvmSpec extends MultiNodeConfig {
+  val seed1 = role("seed1")
+  val seed2 = role("seed2")
+
+  commonConfig(debugConfig(on = false).
+    withFallback(ConfigFactory.parseString("""
+      akka.cluster.auto-down-unreachable-after = 2s
+      akka.cluster.retry-unsuccessful-join-after = 3s
+      akka.remote.retry-gate-closed-for = 45s # setting this 5s will make the test pass
+      akka.remote.log-remote-lifecycle-events = INFO
+                                           """)).
+    withFallback(MultiNodeClusterSpec.clusterConfig))
+
+  class KeepAlive extends Actor {
+    context.system.scheduler.schedule(1.second, 100.millis, self, "tick")(context.dispatcher)
+    Cluster(context.system).subscribe(self, InitialStateAsEvents, classOf[MemberUp])
+    var nodes = Set.empty[Address]
+
+    def receive = {
+      case ClusterEvent.MemberUp(m) ⇒
+        nodes += m.address
+      case "tick" ⇒
+        nodes.foreach { node ⇒
+          context.actorSelection(self.path.toStringWithAddress(node)) ! "ping"
+        }
+      case "ping" ⇒
+    }
+  }
+}
+
+class RestartNode2SpecMultiJvmNode1 extends RestartNode2SpecSpec
+class RestartNode2SpecMultiJvmNode2 extends RestartNode2SpecSpec
+
+abstract class RestartNode2SpecSpec
+  extends MultiNodeSpec(RestartNode2SpecMultiJvmSpec)
+  with MultiNodeClusterSpec with ImplicitSender {
+
+  import RestartNode2SpecMultiJvmSpec._
+
+  @volatile var seedNode1Address: Address = _
+
+  // use a separate ActorSystem, to be able to simulate restart
+  lazy val seed1System = ActorSystem(system.name, system.settings.config)
+
+  def seedNodes: immutable.IndexedSeq[Address] = Vector(seedNode1Address, seed2)
+
+  lazy val restartedSeed1System = ActorSystem(system.name,
+    ConfigFactory.parseString(
+      s"""
+      akka.remote.netty.tcp.port= ${seedNodes.head.port.get}
+      #akka.remote.retry-gate-closed-for = 1s # this is the node that will attempt to re-join
```

Comment:
Remove commented config value and move the comment about the node to the ActorSystem definition.